parser/parse.go

```diff
@@ -261,8 +262,9 @@ func (b *Builder) addDir(dir string, userRequested bool) error {
 	if err != nil {
 		return err
 	}
-	pkgPath := importPathString(buildPkg.ImportPath)
-	if dir != buildPkg.ImportPath {
+	canonicalPackage := canonicalizeImportPath(buildPkg.ImportPath)
+	pkgPath := importPathString(canonicalPackage)
```

Comment:
What about making `canonicalizeImportPath` return type `importPathString`, and explicitly cast to string when needed?